lib/Service/OpenAiAPIService.php

```diff
@@ -434,10 +450,17 @@ public function createChatCompletion(
 				$this->logger->warning('Could not create quota usage for user: ' . $userId . ' and quota type: ' . Application::QUOTA_TYPE_TEXT . '. Error: ' . $e->getMessage(), ['app' => Application::APP_ID]);
 			}
 		}
-		$completions = [];
+		$completions = [
+			'messages' => [],
+			'tool_calls' => [],
+		];
 
 		foreach ($response['choices'] as $choice) {
-			$completions[] = $choice['message']['content'];
+			if ($choice['finish_reason'] === 'tool_calls') {
+				$completions['tool_calls'][] = json_encode($choice['message']['tool_calls']);
+			} else {
+				$completions['messages'][] = $choice['message']['content'];
+			}
```

Comment:
It could be that we have a message and a tool_call in the same choice, I think.

Reply:
The thing is, I think, that a "choice" is one way the model can answer. If we split this up into tool_call choices and messages choices, it's not clear which belong together.

Currently the result is 

```
array{messages: string[], tool_calls: string[]}
```


I think we need something like 

```
array<{message: string, tool_calls: string[]}>
```